js/layout-javascript/small-card-code.js

```diff
@@ -1433,92 +1431,81 @@ DynamicList.prototype.getPermissions = function(entries) {
-
-    allFilters.push(arrangedFilters);
-  });
-
-  filtersData.filters = allFilters
-
   filtersTemplate = Fliplet.Widget.Templates[_this.layoutMapping[_this.data.layout]['filter']];
   var template = _this.data.advancedSettings && _this.data.advancedSettings.filterHTML
-  ? Handlebars.compile(_this.data.advancedSettings.filterHTML)
-  : Handlebars.compile(filtersTemplate());
+    ? Handlebars.compile(_this.data.advancedSettings.filterHTML)
+    : Handlebars.compile(filtersTemplate());
 
   _this.$container.find('.filter-holder').html(template(filtersData));
-}
+};
 
 DynamicList.prototype.filterList = function() {
   var _this = this;
+  var filteredData = [];
   _this.filterClasses = [];
 
   var listData = _this.searchedListItems ? _this.searchedListItems : _this.listItems;
+  var $activeFilterControls = _this.$container.find('.hidden-filter-controls-filter.mixitup-control-active');
 
   if (_this.data.social && _this.data.social.bookmark && _this.mixer) {
     _this.mixer.destroy();
   }
 
   _this.$container.find('.hidden-search-controls').removeClass('no-results');
 
-  if (!$('.hidden-filter-controls-filter.mixitup-control-active').length) {
+  if (!$activeFilterControls.length) {
     _this.$container.find('.new-small-card-list-container').removeClass('filtering');
     _this.isFiltering = false;
     _this.prepareToRenderLoop(listData);
-    _this.renderLoopHTML(function(from, to){
+    return _this.renderLoopHTML(function(from, to){
       _this.onPartialRender(from, to);
     });
-    return;
   }
 
-  $('.hidden-filter-controls-filter.mixitup-control-active').each(function(index, element) {
-    _this.filterClasses.push($(element).data('toggle'));
-  });
-
-  var filteredData = _.filter(listData, function(row) {
-    var filters = [];
-    row.data['flFilters'].forEach(function(obj) {
-      filters.push(obj.data.class);
+  if (_.every($activeFilterControls, function (el) {
+    return el.dataset.key && el.dataset.key.length
+  })) {
+    // Filter UI contains data-filter, i.e. uses new field-based filters
+    var activeFilters = _.mapValues(_.groupBy($activeFilterControls.map(function () {
+      var $elem = $(this);
+      return {
+        key: $elem.data('key'),
+        value: $elem.data('value')
+      };
+    }), 'key'), function (filter) {
+      return _.map(filter, 'value');
     });
 
-    var matched = [];
-    _this.filterClasses.forEach(function(filter) {
-      matched.push(filters.indexOf(filter.toString()) >= 0);
+    filteredData = _.filter(listData, function (record) {
+      return _this.Utils.Record.matchesFilters(record, activeFilters);
+    });
+  } else {
+    // Legacy class-based filters
+    $activeFilterControls.each(function(index, element) {
+      _this.filterClasses.push($(element).data('toggle'));
     });
 
-    // If "_.includes" returns TRUE
-    // we actually want to return FALSE to _.filter
-    return !_.includes(matched, false);
-  });
+    filteredData = _.filter(listData, function(row) {
```

Comment:
is there any chance these functions can be put into utilities so they're not repeated for each layout?

Reply:
I think so, yes. Let me do that.

Reply:
@squallstar Refactored